### PR TITLE
bsunanda:Run2-alca4 (Combined tag for phi symmetry, Dijet, GammaJet, IsoTrack, Muon calibration scheme)

### PR DESCRIPTION
**Pull request by bsunanda**
Combined tag for NZS (phi symmetry), Dijet, GammaJet, IsoTrack, Muon calibration scheme. The "runTheMatrix.py -s --useInput all" yields "10 9 8 7 6 tests passed, 0 1 0 0 0 failed"

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

Hi Sunanda
a failure is not an option for integration.
(may be OK if it was in the reference release already)
Which workflow failed?

It looks like #7514  is included here,
please confirm, so that we can focus on the relevant one.
(It would also be better to have received just one PR instead of ~5 overlapping)

**Comment by slava77** [comment]
Sunanda,

Please update the subject of the PR (the first one-line which now says "bsunanda:Run2-alca4" to be something more descriptive,) we need it for self-documentation so that others can understand what this is.
Thank you.

**Comment by bsunanda** [comment]
I gave a detailed list in the comments. It covers all changes in AlCa. The failure happened in step1 of the test 140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI Step0-PASSED Step1-FAILED Step2-NOTRUN  - time date Tue Feb  3 12:51:45 2015-date Tue Feb  3 12:28:07 2015; exit: 0 35072 0
and I believe that it has nothing to do with the changes in the AlCa code.
________________________________

**Comment by slava77** [comment]
Hi Sunanda,

The PR subject is used in the auto-generated release notes
see, e.g.,

https://github.com/cms-sw/cmssw/releases/CMSSW_7_4_0_pre6

these should be readable without a need to open the full PR page

Please update the subject.

Thank you.

Cheers

		--slava


On 2/3/15 7:26 AM, bsunanda wrote:
> I gave a detailed list in the comments. It covers all changes in AlCa.
> The failure happened in step1 of the test
> 140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI Step0-PASSED
> Step1-FAILED Step2-NOTRUN - time date Tue Feb 3 12:51:45 2015-date Tue
> Feb 3 12:28:07 2015; exit: 0 35072 0
> and I believe that it has nothing to do with the changes in the AlCa code.
> ________________________________
> From: Slava Krutelyov [notifications@github.com]
> Sent: 03 February 2015 13:57
> To: cms-sw/cmssw
> Cc: Sunanda Banerjee
> Subject: Re: [cmssw] bsunanda:Run2-alca4 (#7523)
> 
> 
> Sunanda,
> 
> Please update the subject of the PR (the first one-line which now says
> "bsunanda:Run2-alca4" to be something more descriptive,) we need it for
> self-documentation so that others can understand what this is.
> Thank you.
> 
> —
> Reply to this email directly or view it on
> GitHub<https://github.com/cms-sw/cmssw/pull/7523#issuecomment-72645704>.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/7523#issuecomment-72649457>.
> 

-- 
***********************************************************************
 Vyacheslav (Slava) Krutelyov
 TAMU: Physics Dept Texas A&M MS4242, College Station, TX 77843-4242
 CERN: 42-R-027
 AIM/Skype: siava16              googleTalk: slava77@gmail.com
 (630) 291-5128  Cell (US)       +41 76 275 7116 Cell (CERN)

-----------------------------------------------------------------------

File: Calibration/HcalAlCaRecoProducers/src/AlCaIsoTracksProducer.cc [hunk 1]
```diff
@@ -208,354 +208,274 @@ AlCaIsoTracksProducer::AlCaIsoTracksProducer(const edm::ParameterSet& iConfig)
...
-    }  
-
-  iEvent.put( outputHcalIsoTrackColl, "HcalIsolatedTrackCollection");
-  iEvent.put( outputTColl, "IsoTrackTracksCollection");
-  iEvent.put( outputExTColl, "IsoTrackExtraTracksCollection");
-  iEvent.put( outputEColl, "IsoTrackEcalRecHitCollection");
-  iEvent.put( outputESColl, "IsoTrackPSEcalRecHitCollection");
-  iEvent.put( outputHColl, "IsoTrackHBHERecHitCollection");
-  iEvent.put( outputHOColl, "IsoTrackHORecHitCollection");
+void AlCaIsoTracksProducer::setPtEtaPhi(std::vector< edm::Ref<l1extra::L1JetParticleCollection> >& objref, double &ptL1, double &etaL1, double &phiL1) {
```
Comment (cvuosalo) [inline Calibration/HcalAlCaRecoProducers/src/AlCaIsoTracksProducer.cc]: Can the first parameter of setPtEtaPhi be declared "const"? It looks like it should be, but, at this late stage, don't change this line of code unless other code changes are also necessary.

File: DataFormats/HcalIsolatedTrack/interface/HcalIsolatedTrackCandidate.h [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    std::pair<double,double> EtaPhiEcal() const {
+      return ((etaPhiEcal_) ? std::pair<double,double>(etaEcal_,phiEcal_) : std::pair<double,double>(0,0));
+    }
+    bool etaPhiEcal() const {return etaPhiEcal_;}
+
+    ///eta, phi at HCAL surface
+    void SetEtaPhiHcal(double eta, double phi, int ieta, int iphi) {
+      etaHcal_=eta; phiHcal_=phi; ietaHcal_=ieta; iphiHcal_=iphi; etaPhiHcal_=true;
+    }
+    std::pair<double,double> EtaPhiHcal() const {
```
Comment (cvuosalo) [inline DataFormats/HcalIsolatedTrack/interface/HcalIsolatedTrackCandidate.h]: According to the CMS naming rules, method names should start with a lower case letter. There are seven method names in this class that start with an upper case letter. But this issue is minor -- no change is necessary right now unless other code change are also being made.

**Comment by cvuosalo** [comment]
+1

For #7523 5640522381d1c74dac2f3f69941a65c20abd456a

Updates of several calibration schemes.

Reco code changes are acceptable. 

Jenkins tests [CMSSW_7_4_X_2015-02-03-0200]
(https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7523/2326/summary.html):
compilation, unit tests, matrix tests, static checks are all OK.

As expected, there are no differences compared to the baseline in the
following:

[alternative-comparisons]
(https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_7_4_X_2015-02-03-0200+7523/5878/alternative-comparisons/)
(after stripping false positives like timer data)

and

[validateJR] (https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_7_4_X_2015-02-03-0200+7523/5878/validateJR/)

**Comment by davidlange6** [comment]
closing in 74x - there is the same PR open in 75x (now the development release)

**Pull request closed without merging**